components/apimgt/org.wso2.carbon.apimgt.impl/src/main/java/org/wso2/carbon/apimgt/impl/dao/ApiMgtDAO.java

```diff
@@ -1529,6 +1529,167 @@ public Subscriber getSubscriber(String subscriberName) throws APIManagementExcep
         return subscribedAPIs;
     }
 
+    public Integer getSubscriptionCount(Subscriber subscriber,String applicationName)
+            throws APIManagementException {
+        Integer subscriptionCount = 0;
+        Connection connection = null;
+        PreparedStatement ps = null;
+        ResultSet result = null;
+
+        try {
+            connection = APIMgtDBUtil.getConnection();
+
+            String sqlQuery = "SELECT COUNT(*) AS SUB_COUNT " +
```

Comment:
You can simplify this sql query more.For eg: check getPaginatedSubscribed APIs() included sql query.